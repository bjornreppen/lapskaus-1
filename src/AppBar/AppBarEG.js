import React, { Component } from 'react'
import AppBar from 'material-ui/AppBar'
import IconButton from 'material-ui/IconButton'
import NavigationBack from 'material-ui/svg-icons/navigation/arrow-back'
import ActionSearch from 'material-ui/svg-icons/action/search'
import TaxonSearch from '../TaxonSearch/TaxonSearch'

const styles = {
  title: {
    cursor: 'pointer'
  }
};

export default class AppBarEG extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showSearch: false
    }
  }

  handleSearchButtonClick = () => {
    this.setState({ showSearch: !this.state.showSearch });
  };

  handleSearchClick = taxonId => {
      this.props.onClick(taxonId)
  };

  handleKeyDown = e => {
      if (e.keyCode === 27) {
          this.handleSearchButtonClick();
      }
  };

  render() {
    return (
      <div>
          {this.state.showSearch ? (
              <TaxonSearch
                  onClick={this.handleSearchClick}
                  onAbort={this.handleSearchButtonClick}
                  onKeyDown={this.handleKeyDown}
              />
          ) : (
              <AppBar
                  title={<span style={styles.title}>Artstre</span>}
                  onTitleTouchTap={() => this.props.onClick(0)}
                  iconElementLeft={
                    <IconButton
                        onClick={() => this.props.onClick(1)} // todo: goto parent
                        >
                      <NavigationBack />
                    </IconButton>
                  }
                  iconElementRight={
                    <IconButton onClick={this.handleSearchButtonClick}>
                      <ActionSearch />
                    </IconButton>
                  }
              />
          )}
          </div>
    )
  }
}
