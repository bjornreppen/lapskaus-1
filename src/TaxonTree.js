import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import {List, ListItem} from 'material-ui/List';
import Avatar from 'material-ui/Avatar';
import FileFolder from 'material-ui/svg-icons/file/folder';
import Chip from 'material-ui/Chip';

function TaxonTree (props) {
    return(
        <MuiThemeProvider>
            <List>
                {props.items.length ?
                    props.items.map(item=>
                        <ListItem
                            key={item.Id}
                            leftAvatar={<Avatar icon={<FileFolder />} />}
                            rightIconButton={
                                <Chip>
                                    {item.CumulativeObservationCount}
                                </Chip>
                            }
                            primaryText={item.ScientificName}
                            secondaryText={item.PopularName}
                            onClick={() => props.onClick(item.Id)}
                        />
                    )
                    : <ListItem primaryText="Loading..." />
                }
            </List>
        </MuiThemeProvider>
    )
}

 export default TaxonTree;