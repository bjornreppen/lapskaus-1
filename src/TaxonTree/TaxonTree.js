import React from 'react';
import List, { ListItem, ListItemText, ListItemSecondaryAction } from 'material-ui-next/List';
import Avatar from 'material-ui-next/Avatar';
import FolderIcon from 'material-ui-icons/Folder'
import Chip from 'material-ui-next/Chip';
import backend from '../backend'

function TaxonTree (props) {
    return(
            <List>
                {props.items.length ?
                    props.items.map(item=>
                        <ListItem
                            button
                            key={item.Id}
                            onClick={() => props.onClick(item.Id)}
                        >
                            <Avatar
                                src={backend.getTaxonPhotoUrl(item.Id)}
                            />

                            <ListItemText
                                primary={item.ScientificName}
                                secondary={item.PopularName}
                            />
                            <ListItemSecondaryAction>
                                <Chip
                                    label={item.CumulativeObservationCount}
                                />
                            </ListItemSecondaryAction>
                        </ListItem>
                    )
                    : <ListItem>
                        <ListItemText
                            primary={"Ingen underarter"}
                        />
                    </ListItem>
                }
            </List>
    )
}

 export default TaxonTree;