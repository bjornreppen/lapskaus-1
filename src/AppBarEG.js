import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import NavigationClose from 'material-ui/svg-icons/navigation/close';
import ActionSearch from 'material-ui/svg-icons/action/search';

function handleTouchTap() {
    alert('onClick triggered on the title component');
}

const styles = {
    title: {
        cursor: 'pointer',
    },
};

const AppBarEG = () => (
    <MuiThemeProvider>
        <AppBar
            title={<span style={styles.title}>Artsgrupper</span>}
            onTitleTouchTap={handleTouchTap}
            iconElementLeft={<IconButton><NavigationClose /></IconButton>}
            iconElementRight={<IconButton><ActionSearch /></IconButton>}
        />
    </MuiThemeProvider>
);

export default AppBarEG;