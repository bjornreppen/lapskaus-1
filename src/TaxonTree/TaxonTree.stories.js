import React from 'react';
import { storiesOf } from '@storybook/react';
import TaxonTree from './TaxonTree'
import {muiTheme} from 'storybook-addon-material-ui';

var dummyItems = [
    {
        "id": 1045,
        "aggreggatedCount": 494614,
        "scientificName": "Agaricomycetes",
        "popularName": null
    }];

storiesOf('Taxon Tree', module)
.addDecorator(muiTheme())
.add('default', () => <TaxonTree items={dummyItems} />)
