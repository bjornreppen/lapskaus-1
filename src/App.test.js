import React from 'react';
import AppBarEG from './AppBarEG';
import TaxonTree from './TaxonTree';
import renderer from 'react-test-renderer';

var dummyItems = [
    {
        "Id": 1,
        "TaxonCategory": 0,
        "ScientificNameId": 0,
        "ScientificName": "Animalia",
        "PopularName": "dyreriket",
        "ObservationCount": 8,
        "CumulativeObservationCount": 18588919
    }];

it('renders without crashing', () => {
    expect(renderer.create(<TaxonTree items={dummyItems} />).toJSON()).toMatchSnapshot();
    expect(renderer.create(<AppBarEG />).toJSON()).toMatchSnapshot();
});

