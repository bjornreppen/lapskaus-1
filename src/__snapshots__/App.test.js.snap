// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  style={
    Object {
      "muiPrepared": true,
      "padding": "8px 0px 8px 0px",
    }
  }
>
  <div>
    <span
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "backgroundColor": null,
          "border": 10,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "fontWeight": "inherit",
          "lineHeight": "16px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          style={
            Object {
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 16,
              "paddingLeft": 72,
              "paddingRight": 56,
              "paddingTop": 20,
              "position": "relative",
            }
          }
        >
          <div
            className={undefined}
            size={40}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(188, 188, 188)",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 20,
                "height": 40,
                "justifyContent": "center",
                "left": 16,
                "muiPrepared": true,
                "position": "absolute",
                "top": 16,
                "userSelect": "none",
                "width": 40,
              }
            }
          >
            <svg
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "#ffffff",
                  "display": "inline-block",
                  "fill": "#ffffff",
                  "fontSize": 24,
                  "height": 24,
                  "margin": 8,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
              />
            </svg>
          </div>
          <div
            className={undefined}
            disabled={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "backgroundColor": "rgb(224, 224, 224)",
                "border": 10,
                "borderRadius": 16,
                "boxShadow": null,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "block",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": "inherit",
                "fontWeight": "inherit",
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "padding": 0,
                "position": "absolute",
                "right": 4,
                "textDecoration": "none",
                "top": 12,
                "verticalAlign": null,
                "whiteSpace": "nowrap",
                "width": "fit-content",
              }
            }
            tabIndex={0}
          >
            <span
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontSize": 14,
                  "fontWeight": 400,
                  "lineHeight": "32px",
                  "muiPrepared": true,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "userSelect": "none",
                  "whiteSpace": "nowrap",
                }
              }
            >
              18588919
            </span>
          </div>
          <div
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            Animalia
          </div>
          <div
            style={
              Object {
                "WebkitBoxOrient": null,
                "WebkitLineClamp": null,
                "color": "rgba(0, 0, 0, 0.54)",
                "display": null,
                "fontSize": 14,
                "height": 16,
                "lineHeight": "16px",
                "margin": 0,
                "marginTop": 4,
                "muiPrepared": true,
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "nowrap",
              }
            }
          >
            dyreriket
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;

exports[`renders without crashing 2`] = `
<div
  className={undefined}
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#00bcd4",
      "borderRadius": "0px",
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "flex",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "paddingLeft": 24,
      "paddingRight": 24,
      "position": "relative",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "100%",
      "zIndex": 1100,
    }
  }
>
  <div
    style={
      Object {
        "marginLeft": -16,
        "marginRight": 8,
        "marginTop": 8,
        "muiPrepared": true,
      }
    }
  >
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#ffffff",
              "display": "inline-block",
              "fill": "#ffffff",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </div>
    </button>
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "boxFlex": 1,
        "color": "#ffffff",
        "flex": "1",
        "fontSize": 24,
        "fontWeight": 400,
        "height": 64,
        "letterSpacing": 0,
        "lineHeight": "64px",
        "margin": 0,
        "muiPrepared": true,
        "overflow": "hidden",
        "paddingTop": 0,
        "textOverflow": "ellipsis",
        "whiteSpace": "nowrap",
      }
    }
  >
    <span
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Artstre
    </span>
  </div>
  <div
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": -16,
        "marginTop": 8,
        "muiPrepared": true,
      }
    }
  >
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#ffffff",
              "display": "inline-block",
              "fill": "#ffffff",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;
